pkg/etcd/etcd.go

```diff
@@ -892,3 +891,57 @@ func backupDirWithRetention(dir string, maxBackupRetention int) (string, error)
 	}
 	return backupDir, nil
 }
+
+func GetServerURLFromETCD(ctx context.Context, cfg *config.Control) (string, int, error) {
+	if cfg.Runtime == nil {
+		return "", 0, fmt.Errorf("runtime is not ready yet")
+	}
+	cl, err := GetClient(ctx, cfg.Runtime, endpoint)
+	if err != nil {
+		return "", 0, err
+	}
+	etcdResp, err := cl.KV.Get(ctx, version.Program+"/etcd/apiaddresses")
```

Comment:
Use the exported variable `AddressKey` from pkg/apiaddresses/controller.go

Reply:
fixed